src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -38,7 +38,24 @@ impl AccessChecker for PrivilegeAccess {
         let session = self.ctx.get_current_session();
 
         match plan {
-            Plan::Query { .. } => {}
+            Plan::Query { metadata, .. } => {
+                let metadata = metadata.read().clone();
+                for table in metadata.tables() {
+                    if table.is_source_of_view() {
```

Comment:
Only view needs to check privilege, so `!if table.is_source_of_view()`?

Reply:
umm, non-view tables also need to be checked.

Sometimes just want user to access view-table and can't access non-view tables.